src/main/scala/io/qbeast/spark/index/query/QuerySpecBuilder.scala

```diff
@@ -84,49 +96,54 @@ private[spark] class QuerySpecBuilder(sparkFilters: Seq[Expression])
-
-      val (from, to) =
-        indexedColumns.map { columnName =>
-          // Get the filters related to the column
-          val columnFilters = filters.filter(hasColumnReference(_, columnName))
-
-          // Get the coordinates of the column in the filters,
-          // if not found, use the overall coordinates
-          val columnFrom = columnFilters
-            .collectFirst {
-              case GreaterThan(_, Literal(value, _)) => sparkTypeToCoreType(value)
-              case GreaterThanOrEqual(_, Literal(value, _)) => sparkTypeToCoreType(value)
-              case EqualTo(_, Literal(value, _)) => sparkTypeToCoreType(value)
-              case IsNull(_) => null
-            }
-
-          val columnTo = columnFilters
-            .collectFirst {
-              case LessThan(_, Literal(value, _)) => sparkTypeToCoreType(value)
-              case LessThanOrEqual(_, Literal(value, _)) => sparkTypeToCoreType(value)
-              case EqualTo(_, Literal(value, _)) => sparkTypeToCoreType(value)
-              case IsNull(_) => null
-            }
-
-          (columnFrom, columnTo)
-        }.unzip
-
-      QuerySpace(from, to, revision.transformations)
-    }
+  def extractQuerySpace(rangePredicate: Seq[Expression], revision: Revision): QuerySpace = {
+
+    // Data Filters should not be empty
+    assert(rangePredicate.nonEmpty)
+
+    val indexedColumns = revision.columnTransformers.map(_.columnName)
+
+    val (from, to) =
+      indexedColumns.map { columnName =>
+        // Get the filters related to the column
+        val columnFilters = rangePredicate.filter(hasColumnReference(_, columnName))
+
+        // Get the coordinates of the column in the filters,
+        // if not found, use the overall coordinates
+        val columnFrom = columnFilters
+          .collectFirst {
+            case GreaterThan(_, Literal(value, _)) => sparkTypeToCoreType(value)
+            case GreaterThanOrEqual(_, Literal(value, _)) => sparkTypeToCoreType(value)
+            case LessThan(Literal(value, _), _) => sparkTypeToCoreType(value)
+            case LessThanOrEqual(Literal(value, _), _) => sparkTypeToCoreType(value)
+            case EqualTo(_, Literal(value, _)) => sparkTypeToCoreType(value)
+            case EqualTo(Literal(value, _), _) => sparkTypeToCoreType(value)
+            case IsNull(_) => null
+          }
+
+        val columnTo = columnFilters
+          .collectFirst {
+            case LessThan(_, Literal(value, _)) => sparkTypeToCoreType(value)
+            case LessThanOrEqual(_, Literal(value, _)) => sparkTypeToCoreType(value)
+            case GreaterThan(Literal(value, _), _) => sparkTypeToCoreType(value)
+            case GreaterThanOrEqual(Literal(value, _), _) => sparkTypeToCoreType(value)
+            case EqualTo(_, Literal(value, _)) => sparkTypeToCoreType(value)
+            case EqualTo(Literal(value, _), _) => sparkTypeToCoreType(value)
+            case IsNull(_) => null
```

Comment:
Because of the Literal placement. It can be placed at the left or right of the operator, and it can reflect different from or to clauses. 

Take as an example` id < 3` and `3 < id`. For the same operator (<) would have different cases. 
In the first one (id < 3) we should filter all the values from 3 to Min(id). In the second one (3 < id), we should filter all the values from 3 to Max(id). 

Probably we can transform those filters before to have a more structured an easy way to parse `from` and `to`, at the same time we parse the IN predicate (or other complex ones).

Reply:
Okay, that makes sense! Good catch :)